### PR TITLE
Add new plugin-e2e package

### PR DESCRIPTION
**Pull request by alyssabull**
This PR adds the new `plugin-e2e` package that will eventually replace the `grafana/e2e`package (details in [this PR](https://github.com/grafana/grafana/pull/79969)). Upgrading the tests to the newest package will also address several security issues associated with the `grafana/e2e` package that is currently not being maintained.

### CODE REVIEW

File: playwright/.auth/admin.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (adamyeats) [inline playwright/.auth/admin.json:1]: I think that this file should probably be in `.gitignore`? It contains some sensitive auth stuff that we probably don't want to commit prublicly.

File: .github/workflows/playwright.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    branches: [ main, master ]
+jobs:
+  test:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-node@v4
+      with:
+        node-version: lts/*
```
Comment (adamyeats) [inline .github/workflows/playwright.yml:15]:
```suggestion
        node-version-file: .nvmrc
```

File: playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+const testDirRoot = 'e2e/plugin-e2e/';
+
+export default defineConfig<PluginOptions>({
+  fullyParallel: true,
+  projects: [
+    // Login to Grafana with admin user and store the cookie on disk for use in other tests
+    {
+      name: 'authenticate',
+      testDir: `${dirname(require.resolve('@grafana/plugin-e2e'))}/auth`,
```
Comment (adamyeats) [inline playwright.config.ts:14]: Could be an idea to assign this to a variable at the top of the file, and then assign that variable to `testDIr`, wdyt?

File: playwright/.auth/viewer.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (adamyeats) [inline playwright/.auth/viewer.json:1]: Same with this file, probably an idea to gitignore the `playwright/.auth` folder entirely.

**Review by adamyeats — CHANGES_REQUESTED** [review]
Looks great! I've put this in "Request changes" for now due to the contents of `playwright/.auth` being committed, but after that's resolved feel free to re-request review and I'll approve. 👍

File: playwright/.auth/viewer.json (hunk 4 above)
Comment (aangelisc) [reply playwright/.auth/viewer.json:1]: Can we gitignore the whole `playwright` file just incase?

File: .github/workflows/playwright.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    - name: Install dependencies
+      run: npm install -g yarn && yarn
```
Comment (aangelisc) [inline .github/workflows/playwright.yml:17]: `yarn` should be preinstalled by the `setup-node` action I think, did this fail without this line?

File: src/e2e/plugin-e2e/configEditor.spec.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    configPage.mockHealthCheckResponse({ status: 200 });
+    await page.getByPlaceholder('http://localhost:9090').fill('http://localhost:9090');
+    await expect(
+      configPage.saveAndTest(),
+      formatExpectError('Expected data source config to be successfully saved')
+    ).toBeOK();
+  });
+});
+
+test.describe('test data source with frontend only health check', () => {
```
Comment (aangelisc) [inline src/e2e/plugin-e2e/configEditor.spec.ts:26]: This test isn't related to this data source

File: src/e2e/plugin-e2e/configEditor.spec.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    const configPage = await createDataSourceConfigPage({ type: 'astradb-datasource' });
+    await page.getByPlaceholder('http://localhost:9090').fill('http://localhost:9090');
+    await expect(
+      configPage.saveAndTest(),
+      formatExpectError('Expected save data source config to fail when AstraDB server is not running')
+    ).not.toBeOK();
+  });
+
+  test('valid credentials should return a 200 status code', async ({ createDataSourceConfigPage, page }) => {
+    const configPage = await createDataSourceConfigPage({ type: 'astradb-datasource' });
```
Comment (aangelisc) [inline src/e2e/plugin-e2e/configEditor.spec.ts:16]: Can we update this to test the same case as the cypress test?

**Review by aangelisc — COMMENTED** [review]
One additional comment would be to delete the `cypress` directory and remove the `cypress` dependency. Once that's done and the other comments are addressed I'd say this is good to go 😊

**Review by bossinc — APPROVED** [review]

File: .github/workflows/playwright.yml (hunk 2 above)
Comment (aangelisc) [reply .github/workflows/playwright.yml:15]: @alyssabull I think you missed this one

**Review by aangelisc — CHANGES_REQUESTED** [review]

**Review by aangelisc — APPROVED** [review]

**Review by adamyeats — APPROVED** [review]

**Comment by alyssabull** [comment]
Closes #244

**Pull request merged**